app/views/govuk_publishing_components/components/_step_by_step_nav.html.erb

```diff
@@ -22,6 +22,10 @@
     class="gem-c-step-nav js-hidden <% unless small %>gem-c-step-nav--large<% end %>"
     <%= "data-remember" if remember_last_step %>
     <%= "data-id=#{tracking_id}" if tracking_id %>
+    data-show-text="<%= t("govuk_component.step_by_step_nav.show", default: "Show") %>"
```

Comment:
I wonder if this is the right way to push translations into the views. Not sure what the alternative is though.

I think I've seen this pattern before somewhere:

```html
<script>
var GOVUK.StepByStepTranslations = <%= t("govuk_component.step_by_step_nav").to_json %>;
</script>
```

Then in the actual JS you use `GOVUK.StepByStepTranslations.show` to get the text.

Reply:
I'm not sure I see the benefit of this option. The advantage of the existing approach is that it's tied to the component itself, and we're not adding any new variables outside of the scope of the component JS. 

Also would this work okay in the component guide - wouldn't we be recreating these variables for each instance of the component shown?

Reply:
Yeah I'm not sure as well, I guess the `data-` thing just looks very messy in the code. Perhaps we can refactor that a bit later. Leaving this for now.